go/obscuronode/enclave/sql/edgelessdb.go

```diff
@@ -0,0 +1,480 @@
+		if err != nil {
+			return nil, err
+		}
+
+		// Note: it usually takes around 10-15 seconds for edb to initialise and restart
+		log.Info("Waiting 30 seconds for EDB restart after initialization...")
+		time.Sleep(30 * time.Second)
+	} else if err != nil {
+		// this doesn't happen if the manifest file just didn't exist, maybe there was an IO error
+		return nil, fmt.Errorf("failed to read manifest file - %w", err)
+	}
+
+	// we check that this edgeless DB was initialized with the manifest we expected (which is only known to this enclave)
+	log.Info("Validating edb signature against expected manifest...")
+	err = verifyEdgelessDB(edbCfg.Host, manifest, edbPEM)
+	if err != nil {
+		return nil, err
+	}
+
+	// connect to EDB (standard mysql-type connection, using certificate derived from the CA cert in the manifest)
+	log.Info("Setting up SQL connection to edb...")
+	edbSQL, err := connectToEdgelessDB(edbCfg.Host, edbPEM)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to EdgelessDB - %w", err)
+	}
+
+	// wrap it in our eth-compatible key-value store layer
+	return CreateSQLEthDatabase(edbSQL)
+}
+
+// perform the SGX enclave attestation to verify edb running in a legit enclave and with expected edb version etc.
+func performRemoteAttestation(edbHost string) (string, error) {
+	// we need to make sure this dir exists before we start read/writing files in there
+	err := os.MkdirAll(dataDir, 0o644)
+	if err != nil {
+		return "", err
+	}
+	err = prepareEDBAttestationConf(attestationCfgFilepath)
+	if err != nil {
+		return "", fmt.Errorf("failed to prepare latest edb attestation config file - %w", err)
+	}
+	log.Info("Verifying attestation from edgeless DB...")
+	edbHTTPAddr := fmt.Sprintf("%s:%s", edbHost, edbHTTPPort)
+	certs, tcbStatus, err := GetCertificate(edbHTTPAddr, attestationCfgFilepath)
+	if err != nil {
+		// todo should we check the error type with: err == attestation.ErrTCBLevelInvalid?
+		// for now it's maximum strictness (we can revisit this and permit some tcbStatuses if desired)
+		return "", fmt.Errorf("attestation failed, host=%s, tcbStatus=%s, err=%w", edbHTTPAddr, tcbStatus, err)
+	}
+	if len(certs) == 0 {
+		return "", fmt.Errorf("no certificates found from edgeless db attestation process")
+	}
+
+	log.Info("Successfully verified edb attestation and retrieved certificate.")
+	return string(pem.EncodeToMemory(certs[0])), nil
+}
+
+func prepareEDBAttestationConf(filepath string) error {
+	// This json blob provides confidence in the version of edgeless DB we are talking to.
+	// The latest json for comparison is available here:
+	//     https://github.com/edgelesssys/edgelessdb/releases/latest/download/edgelessdb-sgx.json
+	//
+	// Todo: revisit how we want to enforce this going forwards, but for now I'm just hardcoding the latest json blob:
+	b := []byte("{\n\t\"SecurityVersion\": 2,\n\t\"ProductID\": 16,\n\t\"SignerID\": \"67d7b00741440d29922a15a9ead427b6faf1d610238ae9826da345cea4fee0fe\"\n}")
```

Comment:
@BedrockSquirrel I'd pull this to the top as a constant.

Reply:
I'd say it belongs conceptually as a field in the EdglessDbConfig, and it should be populated from a constant.